buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/ThirdPartyAuditTask.groovy

```diff
@@ -61,21 +82,52 @@ public class ThirdPartyAuditTask extends AntTask {
                 throw new IllegalArgumentException("illegal third party audit exclusion: '" + s + "', wildcards are not permitted!");
             }
         }
-        excludes = classes;
+        excludes = classes.sort();
     }
-    
+
     /**
      * Returns current list of exclusions.
      */
-    public String[] getExcludes() {
+    @Input
+    public List<String> getExcludes() {
         return excludes;
     }
 
+    /**
+     * The jars we're auditing. These are built lazily when first needed so we
+     * don't end up trying to resolve dependencies at configuration time.
+     */
+    @InputFiles
+    public FileCollection getJars() {
+        if (jars != null) {
+            return jars
+        }
+        configuration = project.configurations.findByName('runtime');
+        if (configuration == null) {
+            // some projects apparently do not have 'runtime'? what a nice inconsistency,
+            // basically only serves to waste time in build logic!
+            configuration = project.configurations.findByName('testCompile');
+        }
+        assert configuration != null;
+
+        // we only want third party dependencies.
+        jars = configuration.fileCollection({ dependency ->
```

Comment:
I'm not sure - I didn't want it to fetch dependencies until it had to so I figured this would push it to as late as possible. But I agree it has that "gradle surprise" factor that I wail about from time to time.